org.eclipse.jdt.junit/src/org/eclipse/jdt/internal/junit/ui/TestViewer.java

```diff
@@ -448,10 +448,15 @@ private OpenTestAction getOpenTestAction(TestSuiteElement testSuite) {
 		String testName= testSuite.getTestName();
 		ITestElement[] children= testSuite.getChildren();
 
-		if (testName.startsWith("[") && testName.endsWith("]") && children.length > 0 && children[0] instanceof TestCaseElement) { //$NON-NLS-1$ //$NON-NLS-2$
+		if (children.length > 0 && children[0] instanceof TestCaseElement tce && tce.isDynamicTest()) {
```

Comment:
This doesn't take into account e.g. `@RepeatedTest` that makes effort to pretty-print the test executions as e.g. `repeat 1 of 4` and not just `[1]`, `[2]`, etc..